evalml/pipelines/components/transformers/imputers/time_series_imputer.py

```diff
@@ -194,10 +213,12 @@ def transform(self, X, y=None):
                 y_imputed = y.bfill()
                 y_imputed.pad(inplace=True)
             elif self._impute_target == "interpolate":
-                # TODO: Revert when pandas introduces Float64 dtype
-                y_imputed = y.astype(float).interpolate()
+                y_imputed = y.interpolate()
                 y_imputed.bfill(inplace=True)
-            y_imputed = ww.init_series(y_imputed)
+            # --> https://github.com/alteryx/evalml/issues/4053
+            # by using the y.ww.logical_type, we'll get the downcast type as the final, which is expected and
+            # onyl problematic for boolean which will be floats which is nonsensical
+            y_imputed = ww.init_series(y_imputed, logical_type=y.ww.logical_type)
```

Comment:
Wanted to highlight this behavior - we're avoiding woodwork inference and just using whatever type we downcast to for y. This will essentially match the current behavior in that we can end up with Boolean columns becoming Doubles with non 1s or 0s floating point values imputed for nans, which isn't great but isn't reachable from AutoML search.